front/src/modules/simulationResult/components/SpaceTimeChart/useTrackOccupancy.ts

```diff
@@ -0,0 +1,241 @@
+import { useCallback, useEffect, useMemo, useState } from 'react';
+
+import type { OccupancyZone, Track } from '@osrd-project/ui-charts';
+import { flatMap, forEach, isFunction, keyBy, keys, pick } from 'lodash';
+
+import { osrdEditoastApi } from 'common/api/osrdEditoastApi';
+import { extractEditoastIdFromTrainId } from 'utils/trainId';
+
+import type { PathOperationalPoint, TrainSpaceTimeData } from '../../types';
+import { batchFetch } from './helpers/utils';
+import {
+  getMovableOccupancyZone,
+  getTracksFromZones,
+  type MovableOccupancyZone,
+} from './helpers/zones';
+
+type AsyncState<T> = { type: 'loading'; data?: T; abort?: () => void } | { type: 'ok'; data: T };
+type TracksState = AsyncState<Track[]>;
+type ZonesState = AsyncState<MovableOccupancyZone[]>;
+type OperationalPointState = { selected: boolean; tracks: TracksState; zones: ZonesState };
+
+type DeployedWaypoint = {
+  waypointId: string;
+  operationalPointId: string;
+  operationalPointPosition: number;
+  operationalPointName?: string;
+  zones?: OccupancyZone[];
+  tracks?: Track[];
+};
+
+/**
+ * This hook handles track occupancy zones lifecycle.
+ *
+ * It takes the following inputs:
+ * - infraId
+ * - trains:
+ *   An array with all visible TrainSpaceTimeData items in the SpaceTimeChart
+ * - pathOperationalPoints:
+ *   An array with all visible PathOperationalPoint items along the current path
+ *
+ * It outputs:
+ * - deployedWaypoints:
+ *   An array with all currently deployed waypoints, with their zones, their tracks, and other
```

Comment:
Maybe with that, we can understand better that "deployed" means that its the track occupancy that is deployed
```suggestion
 *   An array with all currently track occupancy deployed waypoints, with their zones, their tracks, and other
```